### PR TITLE
bootterm 0.5 (new formula)

### PR DESCRIPTION
**Pull request by deadbaed**
bootterm is a simple tool to ease communication with serial ports, useful when needing to quickly connect to embedded devices.

bootterm is simple in the sense of having a clear user interface, without the need to learn complicated keyboard shortcuts.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

this is my first time submitting a new formula to homebrew, please let me know if i did something wrong

### CODE REVIEW

File: Formula/b/bootterm.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  homepage "https://github.com/wtarreau/bootterm"
+  url "https://github.com/wtarreau/bootterm/archive/refs/tags/v0.5.tar.gz"
+  sha256 "95cc154236655082fb60e8cdae15823e4624e108b8aead59498ac8f2263295ad"
+  license "MIT"
+  head "https://github.com/wtarreau/bootterm.git", branch: "master"
+
+  depends_on xcode: :build
+
+  def install
+    bin.mkdir
```
Comment (Moisan) [inline Formula/b/bootterm.rb:12]: You don't need to create this directory.
```suggestion

```

File: Formula/b/bootterm.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class Bootterm < Formula
+  desc "Simple, reliable and powerful terminal to ease connection to serial ports"
+  homepage "https://github.com/wtarreau/bootterm"
+  url "https://github.com/wtarreau/bootterm/archive/refs/tags/v0.5.tar.gz"
+  sha256 "95cc154236655082fb60e8cdae15823e4624e108b8aead59498ac8f2263295ad"
+  license "MIT"
+  head "https://github.com/wtarreau/bootterm.git", branch: "master"
+
+  depends_on xcode: :build
```
Comment (Moisan) [inline Formula/b/bootterm.rb:9]: I'm not entirely sure you need this. Could you try without this dependency?

File: Formula/b/bootterm.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  def install
+    bin.mkdir
+    system "make", "PREFIX=#{prefix}", "install"
+  end
+
+  test do
+    system bin/"bt", "-h"
```
Comment (Moisan) [inline Formula/b/bootterm.rb:17]: We prefer test that exercise a bit more features than the help command. How about this?
```suggestion
    assert_match "port", shell_output("#{bin}/bt -l")
```

File: Formula/b/bootterm.rb (hunk 2 above)
Comment (deadbaed) [reply Formula/b/bootterm.rb:9]: Since this is a C project, I thought xcode is required to provide a c compiler in order to build the project.

**Comment by deadbaed** [comment]
Thank you for your feedback @Moisan !

I tested your changes locally, rebuild the formula and ran the tests again, everything is fine. CI seems happy too.

**Review by Moisan — APPROVED** [review]
Thank you @deadbaed for your first contribution! Without people like you, we couldn't run this project.

**Comment by deadbaed** [comment]
Do not forget the maintainers like you  :)

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @deadbaed!

**Pull request merged**